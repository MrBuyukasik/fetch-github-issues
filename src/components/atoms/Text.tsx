import React from 'react';
import {StyleProp, Text as RNText, TextStyle} from 'react-native';
import styled from 'styled-components/native';

const StyledText = styled(RNText)`
  width: 100%;
  font-size: 16px;
`;

interface Props {
  label: string;
  style?: StyleProp<TextStyle> | undefined;
  testID?: string;
}

const Text = (props: Props) => {
  const {label, style, testID} = props;

  return (
    <StyledText testID={testID} style={style}>
      {label}
    </StyledText>
  );
};

export default Text;
