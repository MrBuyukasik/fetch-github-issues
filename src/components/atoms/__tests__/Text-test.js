import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../Text';
fetch = jest.fn(() => Promise.resolve());
const mockFn = jest.fn();

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
