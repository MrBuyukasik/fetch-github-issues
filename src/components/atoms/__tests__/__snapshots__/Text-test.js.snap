// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 16,
        "width": "100%",
      },
    ]
  }
/>
`;
